public/app/features/dashboard-scene/utils/PanelModelCompatibilityWrapper.ts

```diff
@@ -10,10 +10,15 @@ export class PanelModelCompatibilityWrapper implements PanelModel {
   constructor(private _vizPanel: VizPanel) {}
 
   public get id() {
+    // console.log('this._vizPanel.parent', this._vizPanel.parent);
+    // console.log('this._vizPanel', this._vizPanel);
+
     const id = getPanelIdForVizPanel(
```

Comment:
I think we should use `getPanelIdForLibraryVizPanel` instead of modifying `getPanelIdForVizPanel` here. So I think we can check if the vizPanel parent is `LibraryVizPanel` or not and call the appropriate function for it.